examples/pytorch/rgcn/entity_classify_mp.py

```diff
@@ -0,0 +1,446 @@
+        Fanout of each hop starting from the seed nodes. If a fanout is None,
+        sample full neighbors.
+    """
+    def __init__(self, g, target_idx, fanouts):
+        self.g = g
+        self.target_idx = target_idx
+        self.fanouts = fanouts
+
+    """Do neighbor sample
+    Parameters
+    ----------
+    seeds :
+        Seed nodes
+    Returns
+    -------
+    tensor
+        Seed nodes, also known as target nodes
+    blocks
+        Sampled subgraphs
+    """
+    def sample_blocks(self, seeds):
+        blocks = []
+        etypes = []
+        norms = []
+        ntypes = []
+        seeds = th.tensor(seeds).long()
+        cur = self.target_idx[seeds]
+        for fanout in self.fanouts:
+            if fanout is None or fanout == -1:
+                frontier = dgl.in_subgraph(self.g, cur)
+            else:
+                frontier = dgl.sampling.sample_neighbors(self.g, cur, fanout)
+            etypes = self.g.edata[dgl.ETYPE][frontier.edata[dgl.EID]]
+            norm = self.g.edata['norm'][frontier.edata[dgl.EID]]
+            block = dgl.to_block(frontier, cur)
+            block.srcdata[dgl.NTYPE] = self.g.ndata[dgl.NTYPE][block.srcdata[dgl.NID]]
+            block.edata['etype'] = etypes
+            block.edata['norm'] = norm
+            cur = block.srcdata[dgl.NID]
+            blocks.insert(0, block)
+        return seeds, blocks
+
+@thread_wrapped_func
+def run(proc_id, n_gpus, args, devices, dataset):
+    dev_id = devices[proc_id]
+    g, num_of_ntype, num_classes, num_rels, target_idx, \
+        train_idx, val_idx, test_idx, labels = dataset
+
+    node_tids = g.ndata[dgl.NTYPE]
+    sampler = NeighborSampler(g, target_idx, [args.fanout] * args.n_layers)
+    loader = DataLoader(dataset=train_idx.numpy(),
+                        batch_size=args.batch_size,
+                        collate_fn=sampler.sample_blocks,
+                        shuffle=True,
+                        num_workers=args.num_workers)
+
+    # validation sampler
+    val_sampler = NeighborSampler(g, target_idx, [None] * args.n_layers)
+    val_loader = DataLoader(dataset=val_idx.numpy(),
+                            batch_size=args.batch_size,
+                            collate_fn=val_sampler.sample_blocks,
+                            shuffle=False,
+                            num_workers=args.num_workers)
+
+    # validation sampler
+    test_sampler = NeighborSampler(g, target_idx, [None] * args.n_layers)
+    test_loader = DataLoader(dataset=test_idx.numpy(),
+                             batch_size=args.batch_size,
+                             collate_fn=test_sampler.sample_blocks,
+                             shuffle=False,
+                             num_workers=args.num_workers)
+
+    if n_gpus > 1:
+        dist_init_method = 'tcp://{master_ip}:{master_port}'.format(
+            master_ip='127.0.0.1', master_port='12345')
+        world_size = n_gpus
+        th.distributed.init_process_group(backend="nccl",
```

Comment:
should we use sparse embeddings to train for multiple GPUs?

Reply:
Yes, I think we can add this option. (through --spare_emb)